### PR TITLE
docs: add heads up for Prettier users

### LINKED ISSUE #9318: v3 doc: Prettier hasn't supported MDX v2 or v3 yet
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Description

https://docusaurus.io/docs/next/markdown-features/admonitions#usage-with-prettier

Prettier hasn't supported MDX v2 yet.
https://github.com/prettier/prettier/issues/12209

We'll have to advise users to stop formatting MDX by Prettier for a while after upgrading Docusaurus to v3.

What page should we add this admonition?

### Self-service

- [x] I'd be willing to address this documentation request myself.

### PR DESCRIPTION
**Pull request by tats-u**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Motivation

#9318 

### Test links

Deploy preview: https://deploy-preview-_____--docusaurus-2.netlify.app/

## Related issues/PRs

Fix #9318

### CODE REVIEW

File: website/docs/guides/markdown-features/markdown-features-react.mdx [hunk 1]
```diff
@@ -25,6 +25,20 @@ Use the **[MDX playground](https://mdxjs.com/playground/)** to debug them and ma
...
+
+Prettier, the most popular formatter, [supports only the legacy MDX v1](https://github.com/prettier/prettier/issues/12209). If you get an unintentional format result, you may want to add `{/* prettier-ignore */}` before the problem area.
+
+If you are fed up with inserting `{/* prettier-ignore */}` in too many places, you can consider to disable formatting MDX by Prettier by creating the following `.prettierignore` until it starts supporting MDX v2:
+
+```
+*.mdx
+```
+
+[One of the main authors of MDX recommends `remark-cli` with `remark-mdx`](https://github.com/orgs/mdx-js/discussions/2067).
```
Comment (Josh-Cena) [inline website/docs/guides/markdown-features/markdown-features-react.mdx:38]:
````suggestion
Prettier, the most popular formatter, [supports only the legacy MDX v1](https://github.com/prettier/prettier/issues/12209). If you get an unintentional formatting result, you may want to add `{/* prettier-ignore */}` before the problematic area, or add `*.mdx` to your `.prettierignore`, until Prettier has proper support for MDX v2. [One of the main authors of MDX recommends `remark-cli` with `remark-mdx`](https://github.com/orgs/mdx-js/discussions/2067).
````

**Comment by tats-u** [comment]
Changed the version of MDX because of #9451

File: website/docs/migration/v3.mdx [hunk 2]
```diff
@@ -464,6 +464,23 @@ If you created custom Remark or Rehype plugins, you may need to refactor those,
...
+### Formatters
+
+Prettier, the most common formatter, supports only the legacy MDX v1, not v3 yet as of Docusaurus v3.0.0. You can add `{/* prettier-ignore */}` before the incompatible parts of your code to make it work with Prettier.
+
+```mdx
+{/* prettier-ignore */}
+<SomeComponent>Some long text in the component</SomeComponent>
+```
+
+If you get tired of too many `{/* prettier-ignore */}` insertions, you can consider disabling MDX formatting by Prettier by creating the following `.prettierignore` in the project root until it starts supporting MDX v2:
```
Comment (Josh-Cena) [inline website/docs/migration/v3.mdx:476]:
```suggestion
If you get tired of too many `{/* prettier-ignore */}` insertions, you can consider disabling MDX formatting by Prettier by adding the following to your `.prettierignore`, until it starts supporting MDX v3:
```

**Review by Josh-Cena — APPROVED** [review]
I have just been bitten by this recently. I think this helps users more than it harms us, and the wording is currently good enough, so I'm going to merge it. Thank you!

**Comment by tats-u** [comment]
It looks like this PR has not been merged into the 3.1.0 document.

**Pull request merged**